sdk/identity/identity/src/util/subscriptionUtils.ts

```diff
@@ -10,7 +10,9 @@ import { formatError } from "./logging.js";
 export function checkSubscription(logger: CredentialLogger, subscription: string): void {
   if (!subscription.match(/^[0-9a-zA-Z-._ ]+$/)) {
     const error = new Error(
-      "Invalid subscription provided. You can locate your subscription by following the instructions listed here: https://learn.microsoft.com/azure/azure-portal/get-subscription-tenant-id.",
+      `Subscription '${subscription}' contains invalid characters. If this is the name of a subscription, use ` +
+        `its ID instead. You can locate your subscription by following the instructions listed here: ` +
+        `https://learn.microsoft.com/azure/azure-portal/get-subscription-tenant-id`,
```

Comment:
Maybe link to the [TSG section](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/TROUBLESHOOTING.md#troubleshoot-azure-cli-authentication-issues) instead? It shows how to do this with the CLI